### ISSUE TITLE
UTXO's response time problem

### ISSUE THREAD

**Issue by TurgutKanceltik** [dgarage/NBXplorer#300]
Hi every body,

First of all our NBXplorer includes BTC, LTC and DASH. In the early days, when bringing in UTXOs, the response time was pretty short, like 1-2 seconds. But then, as the number of UTXO increased, the response times increased. At first, we got an answer in 40-50 seconds, then it extended up to 2-3 minutes. After that, we applied prune for all strategies and then restarted NBXplorer. This time around, response time has decreased from 2-3 minutes to 30-40 seconds. However, I still think that is a problem because I'm talking about the UTXO number of over 1300-1400. The majority of these UTXOs are very small amounts that we cannot spend. Why can't we spend it because if we spend those small amounts, the transaction size will be huge and the commission amount will be large accordingly.

**Comment by NicolasDorier** [comment]
@TurgutKanceltik 1300/1400 UTXO is not a lot, the time it takes is caused by the number of transactions required to build your wallet UTXO set. Pruning remove the transactions which are not affecting the current UTXO set anymore.

1. Do you run latest version of NBX? 
2. Why not consolidating those small utxos in a single transaction and pay just 1sat/b ?

**Comment by TurgutKanceltik** [comment]
We updated to the latest. However, nothing has changed much, the response still comes in a very long time. Do we really need consolidating those small utxos in a single transaction and pay just 1sat / b? I think we shouldn't need this for this much UTXO.

**Comment by NicolasDorier** [comment]
@TurgutKanceltik have you tried to modify the `--dbcache` parameter? So NBXplorer is trying to keep more data in RAM, this may improve performance?

**Comment by TurgutKanceltik** [comment]
@NicolasDorier Yes we are using it for a long time as `--dbcache 500` but still slow. How do you suggest editing it?

**Comment by NicolasDorier** [comment]
@TurgutKanceltik I think 500 should be plenty enough.
Yeah, I would advise you to spend those small utxo and then prune when it has a bunch of confirmation.
More than 1sat/bytes, as these days the mempool is full.

**Comment by TurgutKanceltik** [comment]
@NicolasDorier  Of course, this is not a problem for me. I can spend small UTXOs. However, I wonder is this a bug? Because the response time is very abnormal for so few UTXO.

**Comment by NicolasDorier** [comment]
I will try to see if I can reproduce. This is indeed strange, though the time it takes is proportional to the number of transactions impacting the UTXO set of the wallet rather than the number of UTXO.
